commcare-cloud/commcare_cloud.py

```diff
@@ -246,8 +246,24 @@ def run(args, unknown_args):
 
 
 def git_branch():
-    return subprocess.check_output("git branch | grep '^*' | cut -d' ' -f2", shell=True,
-                                   cwd=os.path.expanduser('~/.commcare-cloud/ansible')).strip()
+    cwd = os.path.expanduser('~/.commcare-cloud/ansible')
+
+    def get_detached_ref():
+        """
+        :return None if on a local branch, commit or remote branch name otherwise
+        """
+        return subprocess.check_output(
+            "git branch | grep '\* (HEAD detached at .*)' | cut -d' ' -f5 | cut -d')' -f1",
```

Comment:
Interesting using the shell/grep/cut to do regex matching and string manipulation. Python has nice tools in the standard library that do this kind of thing, with (imo) in a more readable syntax. Also, would only need a single subprocess call (`subprocess.check_output("git branch")`) if matching/string manipulation was done in Python.

Reply:
OK fine :)